### ISSUE TITLE
Переход проекта на Open street map. 

### ISSUE THREAD

**Issue by vladimir075** [nicothin/nesia#36]
Добрый день, коллеги. @nicothin @greeceruit 
Николай, ниже, по линку формулировка Олегом небольшой задачи. Прошу помочь нам.
[
](https://docs.google.com/document/d/141c0CeGD_t-7IM-IGialkRG6UJ8tidOp4tHHWhgFbTs/edit?usp=sharing)
Заранее благодарим

**Comment by vladimir075** [comment]
https://docs.google.com/document/d/141c0CeGD_t-7IM-IGialkRG6UJ8tidOp4tHHWhgFbTs/edit?usp=sharing

**Comment by nicothin** [comment]
@greeceruit насколько я понимаю, все технические вопросы - к Олегу.

можем ли мы в ближайшее время созвониться в телеграме, чтоб обсудить и зафиксировать ТЗ?

пока не ясно ЧТО и ГДЕ делать.

могу лишь предположить, что задача - настроить цвета сгруппированных маркеров на некоторой уже внедренной на проекте карте по некоторым критериям.

я в телеге - [@ngromov](https://t.me/ngromov)

**Comment by nicothin** [comment]
итог созвона

Задача 1: Написать стили для сгруппированного маркера. Стили должны включать возможность смены цвета (пока предусмотреть один цвет базовый и два на смену по добавлению класса-модификатора). Стили должны включать возможность смены размера (предусмотреть 3 размера, смена размеров - по аналогии с цветами. Разметка - тег div и текст внутри него.

Задача 2: Написать разметку и стили для одиночного маркера: сам маркер  и текстовой блок на белом фоне (предусмотреть длинный текст в этом блоке) под маркером.

Макета нет.

Визуальный референс:

![image](https://github.com/user-attachments/assets/cc7445ca-726a-4108-b456-38698cc13f50)

Демка планируемого к применению решения: https://leaflet.github.io/Leaflet.markercluster/example/marker-clustering-custom.html

**Comment by greeceruit** [comment]
Да, вроде, всё верно. В задаче 1 нужно предусмотреть вывод числа в кружочек - числа объектов недвижимости в кластере. Само число буду вычислять я в JS. В задаче 2, если мы говорим о растягивающемся (в зависимости от цены) текстовом блоке, то блок должен растягиваться так, чтобы маркер над ним находился по центру блока. Сильно длинного текста в блоке не будет, так как будем использовать буковки K и M для тысяч и миллионов, как это, собственно, происходит и сейчас на terranesia.com

**Comment by nicothin** [comment]
результат в песочнице: https://codepen.io/nicothin/pen/wvLjwme?editors=1100

**Comment by greeceruit** [comment]
> результат в песочнице: https://codepen.io/nicothin/pen/wvLjwme?editors=1100

Супер! А цвет самого единичного маркера задавать как background-color? 

Или можно воспользоваться модификаторами, определёнными для кластеров? (.trns-group-marker--rent)
В общем, разберусь в процессе внедрения....

**Comment by nicothin** [comment]
> > результат в песочнице: https://codepen.io/nicothin/pen/wvLjwme?editors=1100
> 
> Супер! А цвет самого единичного маркера задавать как background-color?
> 
> Или можно воспользоваться модификаторами, определёнными для кластеров? (.trns-group-marker--rent) В общем, разберусь в процессе внедрения....

по БЭМ нельзя брать модификаторы одного блока и использовать на другом.

предполагаю, что цвета те же, что для групповых маркеров.
дописал 3 строки в CSS в песочнице и у одиночных маркеров появились модификаторы с теми же именами и цветами, что у групповых.

**Comment by greeceruit** [comment]
Внедрил....
